eclair-core/src/main/scala/fr/acinq/eclair/wire/protocol/LightningMessageTypes.scala

```diff
@@ -24,6 +24,7 @@ import fr.acinq.eclair.blockchain.fee.FeeratePerKw
 import fr.acinq.eclair.channel.{ChannelFlags, ChannelType}
 import fr.acinq.eclair.{BlockHeight, CltvExpiry, CltvExpiryDelta, Feature, Features, InitFeature, MilliSatoshi, ShortChannelId, TimestampSecond, UInt64}
 import scodec.bits.ByteVector
+import sun.net.util.IPAddressUtil.{isIPv4LiteralAddress, isIPv6LiteralAddress}
```

Comment:
I don't know if that dependency will restrict what JRE we can run on...@pm47 is that ok to use? Otherwise do you know where we can find an equivalent utility function to check if a `String` is a litteral ipv4/ipv6 address? I'm surprised this cannot be found in the common libraries

Reply:
I would use guava's `InetAddresses.isInetAddress`, it's portable and consistent with how we use it:
```scala
case _ if InetAddresses.isInetAddress(host) => IPAddress(InetAddress.getByName(host), port)
```

Reply:
Good catch! looks like a much better function. fixed in [a8e58ee](https://github.com/ACINQ/eclair/pull/2234/commits/a8e58ee76529776b56c432575da0f63a540d872a)

Reply:
The change to `InetAddreses.isInetAddress` required a few changes to correct test failures:
 - [c2df6d2](https://github.com/ACINQ/eclair/pull/2234/commits/c2df6d2fbe710738efae018815d6c3bf71c4ba04) was needed because brackets are not part of a standard IPv6 address *except* when it also includes the port (eg. in a URI). It seems safer to filter the brackets just in-case some node-announcements or API call includes the brackets. Without this change the "get proxy address" test fails [here](https://github.com/remyers/eclair/blob/c2df6d2fbe710738efae018815d6c3bf71c4ba04/eclair-core/src/test/scala/fr/acinq/eclair/tor/Socks5ConnectionSpec.scala#L45) because `useForIPv6 = false` and it would otherwise be interpreted as a DNS hostname.
 - [c2df6d2](https://github.com/ACINQ/eclair/pull/2234/commits/c2df6d2fbe710738efae018815d6c3bf71c4ba04) was needed because the mock api uses `localhost` but the test expects a resolved address `127.0.0.1`. Because `localhost` does not return true for `isInetAddress` it will not be resolved to the default IPv4 localhost address.